ztp/policygenerator/policyGen/policyBuilder_test.go

```diff
@@ -479,3 +480,117 @@ spec:
+      policyName: "gen-sub-policy"
+`
+	// Read in the test PGT
+	pgt := utils.PolicyGenTemplate{}
+	_ = yaml.Unmarshal([]byte(input), &pgt)
+
+	// Set up the files handler to pick up local source-crs and skip any output
+	fHandler := utils.NewFilesHandler("./testData/GenericSourceFiles", "/dev/null", "/dev/null")
+
+	// Run the PGT through the generator
+	pBuilder := NewPolicyBuilder(fHandler)
+	policies, err := pBuilder.Build(pgt)
+
+	// Validate the run
+	assert.Nil(t, err)
+	assert.NotNil(t, policies)
+
+	assert.Contains(t, policies, "test1/test1-gen-sub-policy")
+	policy := policies["test1/test1-gen-sub-policy"].(utils.AcmPolicy)
+	assert.Equal(t, policy.Metadata.Annotations[utils.ZtpDeployWaveAnnotation], "1")
+}
+
+func TestPolicyZtpDeployWaveAnnotationWithMultipleWaves(t *testing.T) {
+	input1 := `
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "test1"
+  namespace: "test1"
+spec:
+  bindingRules:
+    justfortest: "true"
+  sourceFiles:
+    # Create operators policies that will be installed in all clusters
+    - fileName: GenericNamespace.yaml
+      policyName: "gen-policy"
+    - fileName: GenericSubscription.yaml
+      policyName: "gen-policy"
+    - fileName: GenericOperatorGroup.yaml
+      policyName: "gen-policy"
+    - fileName: GenericConfig.yaml
+      policyName: "gen-policy"
+`
+
+	// Read in the test PGT
+	pgt := utils.PolicyGenTemplate{}
+	_ = yaml.Unmarshal([]byte(input1), &pgt)
+
+	// Set up the files handler to pick up local source-crs and skip any output
+	fHandler := utils.NewFilesHandler("./testData/GenericSourceFiles", "/dev/null", "/dev/null")
+
+	// Run the PGT through the generator
+	pBuilder := NewPolicyBuilder(fHandler)
+	policies, err := pBuilder.Build(pgt)
+
+	// Validate the run
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "doesn't match with Policy")
+	assert.NotNil(t, policies)
+
+	input2 := `
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "test1"
+  namespace: "test1"
+spec:
+  bindingRules:
+    justfortest: "true"
+  sourceFiles:
+    # Create operators policies that will be installed in all clusters
+    - fileName: GenericNamespace.yaml
+      policyName: "gen-policy"
+    - fileName: GenericSubscription.yaml
+      policyName: "gen-policy"
+    - fileName: GenericOperatorGroup.yaml
+      policyName: "gen-policy"
+    - fileName: GenericConfigWithoutWave.yaml
+      policyName: "gen-policy"
+`
+	// Read in the test PGT
+	pgt = utils.PolicyGenTemplate{}
+	_ = yaml.Unmarshal([]byte(input2), &pgt)
+
+	// Set up the files handler to pick up local source-crs and skip any output
+	fHandler = utils.NewFilesHandler("./testData/GenericSourceFiles", "/dev/null", "/dev/null")
+
+	// Run the PGT through the generator
+	pBuilder = NewPolicyBuilder(fHandler)
+	policies, err = pBuilder.Build(pgt)
+
+	// Validate the run
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "doesn't match with Policy")
+	assert.NotNil(t, policies)
+}
```

Comment:
Consider adding two more tests

1. If the user has an overlay that changes a wave so that the policy ends up with multiple values
2. If the user has an overlay that "fixes" a mismatched set of waves in the source.

Reply:
Added.